### ISSUE TITLE
CommonJS Support

### ISSUE THREAD

**Issue by wu-joyous** [modelcontextprotocol/typescript-sdk#90]
**Is your feature request related to a problem? Please describe.**
The current SDK is incompatible with "commonjs" packages, such as using @modelcontextprotocol/sdk in Nest.js.

**Describe the solution you'd like**
@modelcontextprotocol/sdk should be compatible with both ESModules and CommonJS.

**Describe alternatives you've considered**
n/a

**Additional context**
n/a

**Comment by jspahrsummers** [comment]
This can be worked around by using dynamic `import()` at runtime. ESM is the way the ecosystem is trending, so we'd prefer to avoid maintaining CJS if possible.

**Comment by wu-joyous** [comment]
Dynamic import does not work when the tsconfig.module is set to commonjs in TypeScript, since "await import" is compiled into "require". It would be highly beneficial to have the build file support both "CJS" and "ESM," similar to packages like [anthropic-sdk-typescript](https://github.com/anthropics/anthropic-sdk-typescript).

**Comment by jspahrsummers** [comment]
You should be able to use `"module": "node16"`, right?

**Comment by Ripididip** [comment]
> **Is your feature request related to a problem? Please describe.** The current SDK is incompatible with "commonjs" packages, such as using @modelcontextprotocol/sdk in Nest.js.
> 
> **Describe the solution you'd like** @modelcontextprotocol/sdk should be compatible with both ESModules and CommonJS.
> 
> **Describe alternatives you've considered** n/a
> 
> **Additional context** n/a

**Comment by lishaduck** [comment]
Yeah, [`commonjs` is essentially deprecated](https://www.typescriptlang.org/docs/handbook/modules/reference.html#commonjs:~:text=You%20probably%20shouldn%E2%80%99t%20use%20this.%20Use%20node16%20or%20nodenext%20to%20emit%20CommonJS%20modules%20for%20Node.js.). You should only ever use `Node16`, `NodeNext`, `Bundler`, `ESXXXX`, and `ESNext`. All other values are heavily discouraged, and exist purely for backwards compatibility.

**Comment by jspahrsummers** [comment]
@wu-joyous Can you try out the CJS version on `main` (credit to @ycjcl868) and see if it works for you?

**Comment by rinormaloku** [comment]
@jspahrsummers I tried it out and it works! Would be cool if we get a release

**Comment by jspahrsummers** [comment]
Done! Released as [1.2.0](https://github.com/modelcontextprotocol/typescript-sdk/releases/tag/1.2.0).
